Core/include/Acts/Navigation/DetectorNavigator.hpp

```diff
@@ -158,6 +168,33 @@ class DetectorNavigator {
     if (nState.currentVolume == nullptr) {
       throw std::invalid_argument("DetectorNavigator: no current volume found");
     }
+
+    // Intersect with the external surfaces
+    // to set up the lookup order
+    for (auto it = state.options.navigation.externalSurfaces.begin();
+         it != state.options.navigation.externalSurfaces.end(); ++it) {
+      auto sIntersection = it->second->intersect(
+          state.geoContext, nState.position, nState.direction,
+          BoundaryTolerance::Infinite(), s_onSurfaceTolerance);
+
+      for (auto& si : sIntersection.split()) {
+        if (si.isValid()) {
+          it->first = si.pathLength();
+        }
+      }
+    }
```

Comment:
I would disencourage doing this in the init because directions change which makes some surfaces intersectable. especially cylinders don't play nice here